### DISCUSSION TITLE
Roadmap and To-Do

### DISCUSSION THREAD

**Discussion by sircharlo** [sircharlo/meeting-media-manager#5260] [General]
Here's a simple tracker of things I'd like to do eventually. I accept code contributions and suggestions!

---

## To-Do / Wishlist

- [ ] Optimize sqlite queries by specifying which columns to fetch

## Done / Implemented

- [x] #4949
- [x] #4855 
- [x] #3479
- [x] Add e2e testing
- [x] For CO week, hide media instead of removing it completely
- [x] Implement past date media folder cleanup functions
- [x] #3653
- [x] #3750
- [x] #2770
- [x] #3481
- [x] Clean up UI based on feedback
- [x] Test migration from previous version at a KH
- [x] Prevent going to day when in past (additional media for example)
- [x] Show warning notification in existing version of M³ about upcoming update
- [x] Fix Chrome image drop
- [x] Fix song adding/loading UI
- [x] Compare existing strings and remove unneeded strings to allow updated translations on Crowdin
- [x] sircharlo/mmm-refactor#59
- [x] sircharlo/mmm-refactor#65
- [x] #3348
- [x] Add documentation website, built using VitePress and deployed to Github Pages
- [x] Make video fetching non-blocking (stream if not downloaded yet)
- [x] Add feature to include meeting songs in the background music
- [x] Fix the audio click when stopping media mid-play
- [x] Add public talk media logic
- [x] #2766 
- [x] Fix/tweak action island popup placement
- [x] Look into macOS full-screen toggle not having the desired behavior
- [x] #2571
- [x] #2302
- [x] #2768
- [x] Cache metadata-extracted thumbnails for faster loading

**Comment by ux-git** [comment]
@sircharlo The docs site shuts down completely? Or will be re-written for the new app version?

**Comment by sircharlo** [comment]
> @sircharlo The docs site shuts down completely? Or will be re-written for the new app version? 

The current docs will be rendered mostly obsolete with the new version. I also find them quite "heavy". 

The new version of M³ is for the most part self-explanatory; perhaps a very simplified docs could replace the existing one; I would prefer to have most essential documentation in the app itself though, with hint text for each setting for example.

**Comment by ux-git** [comment]
Totally agree! Though, some deeper explanation like a full list of supported formats, settings guide, etc would be great to have on a documentation site, to be able to quickly link to it when needed.

Take a look at Docusaurus, it's pretty user-friendly, with nice essential design, and contribution-open. My experience with it is awesome so far. Plus, in the current GitHub pages documentation, when I open it on mobile, all the languages are mixed up with no way to fix it or to find anything. I didn't face this issue with the Docusaurus.

**Comment by sircharlo** [comment]
> Totally agree! Though, some deeper explanation like a full list of supported formats, settings guide, etc would be great to have on a documentation site, to be able to quickly link to it when needed.
> 
> Take a look at Docusaurus, it's pretty user-friendly, with nice essential design, and contribution-open. My experience with it is awesome so far. Plus, in the current GitHub pages documentation, when I open it on mobile, all the languages are mixed up with no way to fix it or to find anything. I didn't face this issue with the Docusaurus.

Very nice!

**Comment by sircharlo** [comment]
@ux-git In about 5 minutes I will release v24.7.1, which implements almost all of the UI/UX improvements you suggested!

Would you mind taking a look when you can to see if some things need to be further tweaked, or if I missed some screens?

Also, here are some dialogs that I didn't see in your mockups:
- Keyboard shortcut picker popup
- Cache clear confirmation popup
- Congregation deletion confirmation popup
- Media custom start/stop picker popup
- Video stop confirmation popup
- Media deletion popup

Do you think you could take a look and offer some feedback to make the UI consistent for these popups as well?

Thanks so much for your help!

Note: I haven't yet added the caption/explainers to all the settings, but they'll be there for the next release. I just didn't want to wait for that to be done before getting your feedback.

**Comment by mrtavor** [comment]
Hi @sircharlo, we added a new page in the project in Figma called “Design review”, where we wrote some of the details along with images to make it clearer. Feel free to leave your additional comments there.

Quick overview.
Most important things are:
• Consistent icons (fill vs stroke & sizes & roundness index)
• Apply min-width and max-width for some elements
• Meeting media items: title, icon, reorder structure and logic.
• Menus and sidebar icon gaps

There are also lower priority points that can be improved and polished. We will mention those in the corresponding priority sections later and will ping you.

**Comment by mrtavor** [comment]
What is the difference between the Media custom start/stop picker popup and Video stop confirmation popup?

**Comment by sircharlo** [comment]
> What is the difference between the Media custom start/stop picker popup and Video stop confirmation popup?

The video stop popup is what shows up when attempting to stop a video while it is playing; to prevent accidentally stopping a song for example.

The custom start-stop modal is what shows up when you click on the black HH:MM badge on a video. It allows you to set a custom start and stop time for the video (useful for when a congregation is playing a streamed public talk, they can set the start and stop times without having to cut the actual video file).

**Comment by sircharlo** [comment]
@ux-git @mrtavor I just released 24.7.2 in which the vast majority of the improvements you suggested are implemented. 
I documented a few icons that appear to be missing from the UI kit, and a few for which I could not find a use. 
When you have time, I’d love to see your thoughts on what could be still be improved or changed, or things I might have overlooked. 
Thanks again!

**Comment by alanfermtz** [comment]
Based on what I have been trying is really good. Its feels smoother.
And thanks for adding zoom in to jw org mirroring, its perfect!

Also, worth mentioning its that now I see more often images on both apps (stable, refactor) that are related to watchtower footnotes, and also the brochure love people. Its a bug? If its intended, can you make a exclude button for this cases?

**Comment by sircharlo** [comment]
> Based on what I have been trying is really good. Its feels smoother. And thanks for adding zoom in to jw org mirroring, its perfect!

Thanks so much!

> Also, worth mentioning its that now I see more often images on both apps (stable, refactor) that are related to watchtower footnotes, and also the brochure love people. Its a bug? If its intended, can you make a exclude button for this cases?

Did you look into these options in the advanced settings?

![image](https://github.com/user-attachments/assets/ba84320e-9130-4399-a642-6c6f017c90a4)

**Comment by alanfermtz** [comment]
Yes, I have that enabled
I still see all this images:

![IMG-20240808-WA0029](https://github.com/user-attachments/assets/34b274b3-b031-40af-b11f-1d54491d9467)
![IMG-20240808-WA0030](https://github.com/user-attachments/assets/f46e25b7-40da-44e3-8882-0c5e654350c3)

**Comment by sircharlo** [comment]
> Yes, I have that enabled I still see all this images:

Those two pictures are from talk 1. “Tell the Next Generation” About God’s Power, specifically the reference: [w14 1/15 23 ¶4-5](https://wol.jw.org/en/wol/pc/r1/lp-e/202024246/3/0). 

In 2014, the Spanish Watchtower publication files were structured differently, which makes it more difficult to know which paragraph a picture is for. For this reason, all pictures are included.

In other languages, such as English, those pictures don't show up because the publication files are structured in a way that makes it clear which paragraph media relates to.

**Comment by ux-git** [comment]
> @ux-git @mrtavor I just released 24.7.2 in which the vast majority of the improvements you suggested are implemented. I documented a few icons that appear to be missing from the UI kit, and a few for which I could not find a use. When you have time, I’d love to see your thoughts on what could be still be improved or changed, or things I might have overlooked. Thanks again!

Great! Sure, we will take a look, most likely starting from Monday.

**Comment by mtdvlpr** [comment]
> Totally agree! Though, some deeper explanation like a full list of supported formats, settings guide, etc would be great to have on a documentation site, to be able to quickly link to it when needed.
> 
> Take a look at Docusaurus, it's pretty user-friendly, with nice essential design, and contribution-open. My experience with it is awesome so far. Plus, in the current GitHub pages documentation, when I open it on mobile, all the languages are mixed up with no way to fix it or to find anything. I didn't face this issue with the Docusaurus.

I would personally recommend [VitePress](https://vitepress.dev/), because it's Vue-based, just like M3 itself. It's very easy to setup and extremely fast!

**Comment by mtdvlpr** [comment]
> Hi @sircharlo, we added a new page in the project in Figma called “Design review”, where we wrote some of the details along with images to make it clearer. Feel free to leave your additional comments there.

Any chance I can be added to that Figma file? I would love to see it and perhaps even contribute my thoughts on it.

**Comment by ux-git** [comment]
> Add a re-fetch button (in settings?) to force a refresh of scheduled media

@sircharlo  What about adding the re-fetch button right in the Media sync window? For example one button for all ERROR media. This way it's really intuitive and allows to solve the problem right away _[by any brother at the computer]_ instead of investigating into the settings options _[and asking the specific brother who knows where to look at]_:

<img width="498" alt="Screenshot 2024-12-02 at 04 54 28" src="https://github.com/user-attachments/assets/18b7b683-4b65-43c3-8ea8-a77801f36408">

**Comment by ux-git** [comment]
> What about adding the re-fetch button right in the Media sync window?

1. Additionally, to solve the missing media anxiety in cases when re-initiating the sync won't help (e.g. media not yet available), M³ could show the meeting date for each media in the 'errors' section.Either on hover or with a smaller description-like text underneath.

This will let users see that the error is not for today's meeting, but rather a few weeks later, so they won't have to worry. Plus, the media naming is not humanly-readable, making it harder to troubleshoot errors by searching for a iifa12-p2.mp4 in the Library for example. But having the date will allow you to simply go to that date on a mobile device, and find out that one of the media items is not yet available.

2. Also, maybe some kind of "Try later" button would make sense in this setup. Let's say, a missing media is not yet available and it's 4 weeks away. Then by clicking "Try later" the media will try to sync in 7 days once again, but at least it won't show the error state on the sync button, giving more peace of mind to AV operators.

**Comment by sircharlo** [comment]
> > What about adding the re-fetch button right in the Media sync window?
> 
> 1. Additionally, to solve the missing media anxiety in cases when re-initiating the sync won't help (e.g. media not yet available), M³ could show the meeting date for each media in the 'errors' section.Either on hover or with a smaller description-like text underneath.
> 
> This will let users see that the error is not for today's meeting, but rather a few weeks later, so they won't have to worry. Plus, the media naming is not humanly-readable, making it harder to troubleshoot errors by searching for a iifa12-p2.mp4 in the Library for example. But having the date will allow you to simply go to that date on a mobile device, and find out that one of the media items is not yet available.
> 
> 2. Also, maybe some kind of "Try later" button would make sense in this setup. Let's say, a missing media is not yet available and it's 4 weeks away. Then by clicking "Try later" the media will try to sync in 7 days once again, but at least it won't show the error state on the sync button, giving more peace of mind to AV operators.

Let me know what you thnk of the changes to the download popup in the next beta!
